src/app/login/login-form.component.ts

```diff
@@ -22,7 +22,7 @@ const repeatPassword = {
 };
 
 const loginForm = {
-  name: [ '', [ Validators.required, Validators.pattern(/^[a-z0-9]+$/i) ] ],
+  name: [ '', [ Validators.required, Validators.pattern(/^[a-z0-9_.-]+$/i) ] ],

```

Comment:
We will allow even `@` on login on parent. And we will just need validation for registration. We can skip validator on login.